src/js/player.js

```diff
@@ -312,6 +312,34 @@ class Player extends Component {
     // Run base component initializing with new options
     super(null, options, ready);
 
+    // we don't want a source change for the first source set
+    let sourceSet = false;
+
+    if (tag.src) {
+      sourceSet = true;
+    }
+    const triggerSourceChange = () => {
+      if (!sourceSet) {
+        sourceSet = true;
+        return;
+      }
+      this.trigger('sourcechange');
+    };
+
+    if (window.MutationObserver) {
```

Comment:
for now we use `MutationObserver`, but there has got to be another way that is more portable across all the browsers.

Reply:
we can definitely say it's only available on browsers where MutationObserver is supported for this new feature given the costs of supporting it for all browsers.